src/api/app/models/workflow/step.rb

```diff
@@ -77,6 +77,10 @@ def target_package_name
 
   private
 
+  def target_project_base_name
+    raise NotImplementedError
```

Comment:
We should raise `AbstractMethodCalled` instead of `NotImplementedError`:

```suggestion
    raise AbstractMethodCalled
```

 `NotImplementedError` is for platform limitations like if the OS doesn't support a certain feature. This is documented [here](https://ruby-doc.org/core-2.5.0/NotImplementedError.html).

Reply:
done